### ISSUE TITLE
How do I retrieve an array element that was removed from a class inside afterSave?

### ISSUE THREAD

**Issue by srameshr** [ParsePlatform/Parse-SDK-JS#190]
```
// Define
Parse.Cloud.define('name', function() {
 // del an item from array
 someClass.remove("users", userToRemove); //  How to access this userToRemove inside afterSave
 someClass.set("updateType", "removeUser");
});

// AfterSave
Parse.Cloud.afterSave("Class", function(request) {
  if(request.object.get("updateType") === "removeUser") {
    // How to access that a  array element that was removed earlier in here
})

```

**Comment by andrewimm** [comment]
Unfortunately, the changes made to an object are only available in the beforeSave hook, not in the afterSave hook. I believe this info might be retrievable from the over-the-wire format we use for custom webhooks on third-party servers, but not within Cloud Code.

**Comment by srameshr** [comment]
So, is it possible to obtain the ```data``` object that is sent to a cloud function in inside beforeSave? Something like:

```
Parse.Cloud.run("foo", {
  'id': '123456789'
});

Parse.Cloud.beforeSave('someClass', function(request, response) {
  var id = request.params.id;
  // This is undefined. Is there any way to obtain this param inside beforeSave
})

```

**Comment by wangmengyan95** [comment]
Hi @srameshr, `beforeSave` will only be triggered when you save a `ParseObject`. You cloud functions will get the request.params but the bfreoSave will not.

**Comment by srameshr** [comment]
@wangmengyan95  @andrewimm  Yeah got it. I kind of created a new column that holds the element that was removed and then run afterSave and do some processing on this element and remove it from the new column. Very frisky but works.

Is there are better solution to do something like this within the cloud? 

And one more thing is, triggering a particular function inside afterSave is quite difficult. As of now I manually set ```updateType```  column in ```Parse.Cloud.define`` and then inside ```afterSave``` I check for the value of updateType on the request object, i.e:
```
if(request.object.get('updateType') === 'somethig') {
  // Do something
}
else if(request.object.get('updateType') === 'somethingElse') {
  // Do something else
}

```

Is it possible to specify  a function  name that has to be run in ```afterSave``` during regular object update inside ```Parse.Cloud.define```
